### PR TITLE
Fixed #24299 -- Fixed ContentType creation error message

### PR DESCRIPTION
**Pull request by sir-sigurd**

### CODE REVIEW

**Comment by timgraham** [comment]
Could you please add a test?

**Comment by sir-sigurd** [comment]
test added

File: django/contrib/contenttypes/tests/tests.py [hunk 1]
```diff
@@ -264,3 +265,26 @@ def test_name_deprecation(self):
...
+    def test_message_if_get_for_model_fails(self, mocked_get, mocked_get_or_create):
+        """
+        Check that `RuntimeError` with nice error message is raised if
+        `get_for_model` fails because of database errors.
+        """
+
+        def _test_message(mocked_method):
+            for ExceptionClass in (IntegrityError, OperationalError, ProgrammingError):
+                mocked_method.side_effect = ExceptionClass
+                with self.assertRaisesRegexp(
```
Comment (timgraham) [inline django/contrib/contenttypes/tests/tests.py]: assertRaisesRegexp is deprecated on Python 3, you use `self.assertRaisesMessage` I think

**Comment by sir-sigurd** [comment]
updated

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Seems like it might be simpler to update the auth migration to depend on the second contenttypes migration. Did you try this?
```diff
diff --git a/django/contrib/auth/migrations/0001_initial.py b/django/contrib/aut
index a58f73a..a20140a 100644
--- a/django/contrib/auth/migrations/0001_initial.py
+++ b/django/contrib/auth/migrations/0001_initial.py
@@ -10,7 +10,7 @@ from django.utils import timezone
 class Migration(migrations.Migration):
 
     dependencies = [
-        ('contenttypes', '__first__'),
+        ('contenttypes', '0002_remove_content_type_name'),
     ]
 
     operations = [
```

**Comment by MarkusH** [comment]
I'd rather not change an existing migration but have an empty migration that just depends on `contenttypes.0002` and `auth.0005` (in that order). That's what `makemigrations --merge` does but not on app-level but cross apps. I think that's the cleaner solution to the problem (which is actually the same as in [#24075](https://code.djangoproject.com/ticket/24075)).

Draft:
```python
class Migration():
    dependencies = [
        ('contenttypes', '0001.....'),
        ('auth', '0005.....'),
    ]

**Comment by sir-sigurd** [comment]
@timgraham 
This will fix `auth` app, but I see no reason why users can't use `ContentType` in their own post_migrate handlers.

I think that code fragment I changed in my PR should either be fixed or removed anyway.

**Comment by MarkusH** [comment]
@sir-sigurd Of course people can use `ContentType` in their `post_migrate` signal, but as of now there is no way to make sure the signal is only fired for an app if that app is fully migrated. I'm working on a fix for 1.9 though.

**Comment by timgraham** [comment]
merged in d392c1e150a46aca72e37263c44c034d92a79f79, thanks. Also created https://github.com/django/django/pull/4150 to help with this issue.

**Pull request closed without merging**